File: chat-gui/src/main/java/ru/usedesk/chat_gui/internal/chat/TextChangeListener.java
package ru.usedesk.chat_gui.internal.chat;

import android.text.Editable;
import android.text.TextWatcher;

import androidx.annotation.NonNull;

class TextChangeListener implements TextWatcher {

    private final TextChangedAction textChangedAction;

    TextChangeListener(@NonNull TextChangedAction textChangedAction) {
        this.textChangedAction = textChangedAction;
    }

    @Override
    public void beforeTextChanged(CharSequence s, int start, int count, int after) {

    }

    @Override
    public void onTextChanged(CharSequence s, int start, int before, int count) {
        textChangedAction.onTextChanged(s.toString());
    }

    @Override
    public void afterTextChanged(Editable s) {

    }

    interface TextChangedAction {
        void onTextChanged(@NonNull String text);
    }
}

File: chat-gui/src/main/java/ru/usedesk/chat_gui/chat/adapters/TextChangeListener.kt
package ru.usedesk.chat_gui.chat.adapters

import android.text.Editable
import android.text.TextWatcher

internal class TextChangeListener(
        private val onTextChanged: (String) -> Unit
) : TextWatcher {

    override fun beforeTextChanged(s: CharSequence, start: Int, count: Int, after: Int) {}

    override fun onTextChanged(s: CharSequence, start: Int, before: Int, count: Int) {
        onTextChanged(s.toString())
    }

    override fun afterTextChanged(s: Editable) {}
}